app/controllers/products_controller.rb

```diff
@@ -1,5 +1,6 @@
 class ProductsController < ApplicationController
   before_action :find_individual_product, only: [:show, :edit, :update, :retire]
+  before_action :require_login, only: [:new, :create, :edit, :update, :destroy]
```

Comment:
I think the last symbol should be :retire instead of :destroy unless it was decided we're also destroying a product!

Reply:
done! will push my changes in 2 mins